core/services/workflows/models.go

```diff
@@ -85,7 +86,21 @@ type step struct {
 type triggerCapability struct {
 	workflows.StepDefinition
 	trigger capabilities.TriggerCapability
-	config  *values.Map
+
+	config *values.Map
+	mu     sync.RWMutex
+}
```

Comment:
How about an [atomic.Pointer](https://pkg.go.dev/sync/atomic#Pointer)?
```suggestion
	config atomic.Pointer[values.Map]
```